jgrapht-core/src/main/java/org/jgrapht/graph/interval/IntervalGraphMapping.java

```diff
@@ -0,0 +1,354 @@
+        this.mappingLive = false;
+    }
+
+    /**
+     * Basic constructor for a live mapping to an interval graph. All methods are passed through to the underlying interval graph.
+     * Every vertex and edge update will be considered.
+     *
+     * @param graph the graph to be mapped
+     */
+    private IntervalGraphMapping(ListenableGraph<V, E> graph) {
+        this((Graph<V, E>) graph);
+        // add the basic IntervalGraphListener to the ListenableGraph such that we have a live mapping
+        graph.addGraphListener(new IntervalGraphMappingListener<>(this));
+        // as we know that a ListenableGraph is used, we set the mapping live
+        this.mappingLive = true;
+    }
+
+    /**
+     * Constructs a new live interval graph mapping with a new graph. The graph can either be directed or undirected, depending on the
+     * specified edge supplier.
+     *
+     * @param vertexSupplier the vertex supplier of the new graph
+     * @param edgeSupplier the edge supplier of the new graph.
+     * @param weighted whether the graph is weighted, i.e. the edges support a weight attribute
+     */
+    public IntervalGraphMapping(Supplier<V> vertexSupplier, Supplier<E> edgeSupplier, boolean weighted) {
+        this(new DefaultListenableGraph<>(new DefaultUndirectedGraph<>(vertexSupplier, edgeSupplier, weighted)));
+    }
+
+    /**
+     * Construct a new live interval graph mapping with a new graph with given <code>vertices</code>. The graph can either be directed or undirected, depending on the
+     * specified edge supplier.
+     * Runs in linear time if vertex list is sorted by the starting point and the end point of the interval, otherwise naive insertion is performed
+     *
+     * @param vertices initial vertices
+     * @param vertexSupplier the vertex supplier of the new graph.
+     * @param edgeSupplier the edge supplier of the new graph.
+     * @param weighted whether the graph is weighted, i.e. the edges support a weight attribute
+     *
+     * @throws NullPointerException if the specified edge factory is <code>
+     * null</code>.
+     */
+    public IntervalGraphMapping(ArrayList<V> vertices, Supplier<V> vertexSupplier, Supplier<E> edgeSupplier, boolean weighted) {
+        this(vertexSupplier, edgeSupplier, weighted);
+
+        boolean isSorted = true;
+        V current = vertices.get(0);
+        for (int i = 1; i < vertices.size(); i++) {
+            V next = vertices.get(i);
+            if (current.getInterval().compareTo(next.getInterval()) > 0) {
+                isSorted = false;
+                break;
+            }
+            current = next;
+        }
+
+        if(isSorted) {
+            ArrayList<LinkedList<V>> edges = new ArrayList<>(vertices.size());
+            ArrayList<Interval<T>> vertexIntervals = new ArrayList<>(vertices.size());
+
+            // calculate all edges
+            for(int i = 0; i < vertices.size(); ++i) {
```

Comment:
You should probably use the standard sort algorithms of ArrayList (and maybe the comparatos, which I added for the interval nice decomposition). With a list of sorted intervals by ending point and a list by starting points, this would reduce the complexity from O(|V|^2) to O(|V| log (|V|))

Reply:
We should reduce it to |E| and not |V| log |V|. But you are correct that this is not optimally implemented.

Reply:
It should get at least |E| + |V| log |V|, yes. It was late, when I made the review x)

Reply:
I fixed that